packages/patternfly-4/react-core/src/experimental/components/DataToolbar/DataToolbarUtils.tsx

```diff
@@ -0,0 +1,64 @@
+import { getModifier } from '@patternfly/react-styles';
+import styles from '@patternfly/react-styles/css/components/DataToolbar/data-toolbar';
+import * as React from 'react';
+import { RefObject } from 'react';
+
+interface DataToolbarContextProps {
+  isExpanded: boolean;
+  toggleIsExpanded: () => void;
+  expandableContentRef: RefObject<HTMLDivElement>;
+  expandableContentId: string;
+}
+
+export const DataToolbarContext = React.createContext<Partial<DataToolbarContextProps>>({});
+
+export type DataToolbarBreakpointMod = {
+  /** The attribute to modify  */
+  modifier: 'hidden' | 'visible' | 'align-right' | 'align-left';
+  /** The breakpoint at which to apply the modifier */
+  breakpoint: 'md' | 'lg' | 'xl' | '2xl';
+};
+
+export const formatBreakpointMods = (breakpointMods: DataToolbarBreakpointMod[]) => {
+
+  let formattedBreakpointMods = '';
+  for (const breakpointMod of breakpointMods) {
+    formattedBreakpointMods += `${getModifier(styles, `${breakpointMod.modifier}-on-${breakpointMod.breakpoint}`)} `;
+  }
+  return formattedBreakpointMods;
+};
+
+export type DataToolbarSpacer = {
+  /** The size of the spacer */
+  spacerSize: 'none' | 'sm' | 'md' | 'lg';
+  /** The breakpoint at which to apply the spacer */
+  breakpoint?: 'md' | 'lg' | 'xl';
+};
+
+export const formatItemSpacers = (spacers: DataToolbarSpacer[]) => {
+
+  let formattedSpacers = '';
+  for (const spacer of spacers) {
```

Comment:
Same as above, this can be rewritten to reduce
```JS
return spacers.reduce((acc, curr) => (
  `${acc} pf-m-spacer-${curr.spacerSize}${curr.breakpoint ? `-on-${curr.breakpoint}` : ''}`
), '')
```